code/obj/item/sec_tape.dm

```diff
@@ -0,0 +1,140 @@
+#define MAXTAPE 40
+#define STARTAMOUNT 1
+/// The security tape roll object, used to create security tapes.
+/obj/item/sec_tape
+	name = "security tape roll"
+	var/base_name = "security tape roll"
+	desc = "A roll of security tape."
+	amount = STARTAMOUNT
+	max_stack = MAXTAPE
+	stack_type = /obj/item/sec_tape
+	icon = 'icons/obj/sec_tape.dmi'
+	icon_state = "sec_tape_roll"
+	w_class = W_CLASS_TINY
+	flags = TABLEPASS|EXTRADELAY|FPRINT|CONDUCT|ONBELT
+	rand_pos = 1
+	event_handler_flags = USE_GRAB_CHOKE | USE_FLUID_ENTER
+	special_grab = /obj/item/grab
+	inventory_counter_enabled = 1
+
+	New(loc, length = STARTAMOUNT)
+		src.amount = length
+		..(loc)
+		BLOCK_SETUP(BLOCK_ROPE)
+
+	before_stack(atom/movable/O, mob/user)
+		user.visible_message("<span class='notice'>[user] begins awkwardly sticking the tape back onto the roll.</span>")
+
+	after_stack(atom/movable/O, mob/user, var/added)
+		boutput(user, "<span class='notice'>You finish sticking the tape back together.</span>")
+
+	change_stack_amount(diff)
+		. = ..()
+		tooltip_rebuild = 1
+		src.UpdateIcon()
+
+	custom_suicide = 1
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		//Are we on valid turf?
+		var/turf/T = get_turf(user)
+		if (isturf(user.loc) && istype(T, /turf/simulated/floor) && amount >= 4)
+			//Setup the perfect crime
+			user.visible_message("<span class='alert'><b>[user] carefully sets up some security tape around themselves then swallows the tape and chokes on it, satisfied that they created the perfect crime scene.</b></span>")
+			var/obj/sec_tape/T1 = new /obj/sec_tape(T, src)
+			T1.set_dir(EAST)
+			var/obj/sec_tape/T2 = new /obj/sec_tape(T, src)
+			T2.set_dir(WEST)
+			var/obj/sec_tape/T3 = new /obj/sec_tape(T, src)
+			T3.set_dir(NORTH)
+			var/obj/sec_tape/T4 = new /obj/sec_tape(T, src)
+			T4.set_dir(SOUTH)
+			//No need to set it up to the south, it already is by default
+			user.death(FALSE)
+			qdel(src)
+		else
+			//Else we'll choke ourselves out
+			user.visible_message("<span class='alert'><b>[user] wraps some tape around \his neck and tightens it.</b></span>")
+			user.take_oxygen_deprivation(160)
+			SPAWN(50 SECONDS)
+				if (user && !isdead(user))
+					user.suiciding = 0
+		return 1
+
+	update_icon()
+		inventory_counter?.update_number(amount)
+
+/obj/item/sec_tape/vended //Tape in secvends has a pre-set amount of uses
+	New(loc)
+		..(loc)
+		amount = 20
```

Comment:
If you aren't passing the `amount` as an argument (which was the sensible way to do it imo, not sure why you changed it) then you will need to call UpdateIcon here.

Reply:
Done so